File: password.cs
using System;

namespace PasswordGenerator
{
    class Program
    {
        static Random rand = new Random();

        static string GeneratePassword(int length, bool useDigits, bool useSpecialChars)
        {
            string characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
            if (useDigits)
            {
                characters += "0123456789";
            }
            if (useSpecialChars)
            {
                characters += "!@#$%^&*()-_=+[]{}|;:,.<>?";
            }

            char[] password = new char[length];
            for (int i = 0; i < length; i++)
            {
                int randomIndex = rand.Next(characters.Length);
                password[i] = characters[randomIndex];
            }

            return new string(password);
        }

        static void Main(string[] args)
        {
            Console.Write("Enter desired password length: ");
            int passwordLength;
            if (!int.TryParse(Console.ReadLine(), out passwordLength) || passwordLength <= 0)
            {
                Console.WriteLine("Invalid input. Using default password length of 12.");
                passwordLength = 12;
            }

            string password = GeneratePassword(passwordLength, true, true);
            Console.WriteLine("Generated Password: " + password);
        }
    }
}

File: password.dart
import 'dart:io';
import 'dart:math';

String generatePassword(int length, bool useDigits, bool useSpecialChars) {
  String characters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
  if (useDigits) {
    characters += "0123456789";
  }
  if (useSpecialChars) {
    characters += "!@#\$%^&*()-_=+[]{}|;:,.<>?";
  }

  String password = '';
  Random random = Random();
  for (int i = 0; i < length; i++) {
    int randomIndex = random.nextInt(characters.length);
    password += characters[randomIndex];
  }

  return password;
}

void main() {
  stdout.write("Enter desired password length: ");
  String? input = stdin.readLineSync();
  int passwordLength = 12; // Default length

  try {
    passwordLength = int.parse(input!);
  } catch (e) {
    print("Invalid input. Using default password length of 12.");
  }

  String password = generatePassword(passwordLength, true, true);
  print("Generated Password: $password");
}
